src/sentry/notifications/platform/provider.py

```diff
@@ -25,12 +27,30 @@ class NotificationProvider[RenderableT](Protocol):
     """
 
     key: NotificationProviderKey
-    default_renderer: type["NotificationRenderer[RenderableT]"]
-    target_class: type["NotificationTarget"]
+    default_renderer: NotificationRenderer[RenderableT]
+    target_class: type[NotificationTarget]
     target_resource_types: list[NotificationTargetResourceType]
 
     @classmethod
-    def validate_target(cls, *, target: "NotificationTarget") -> None:
+    def dispatch_notification(
+        cls,
+        *,
+        target: NotificationTarget,
+        template: Any,
+        data: Any,
+        thread_id: str | None = None,
+    ) -> None:
+        renderer = cls.get_renderer(notification_type=template.notification_type)
+        notification_content = renderer.render(data=data, template=template)
+        cls._send_notification(
+            notification_content=notification_content,
+            notification_type=template.notification_type,
+            target=target,
+            thread_id=thread_id,
+        )
+
```

Comment:
This allows us to sidestep the generic type ambiguity issues we saw before, since we can guarantee the `RenderableT` typing here via the `cls` type.